### ISSUE TITLE
[clang-tidy] modernize-type-traits false positive

### ISSUE THREAD

**Issue by vient** [llvm/llvm-project#67896]
https://godbolt.org/z/b537zPjo9
```cpp
namespace std {

template <class a>
struct disjunction : a {};

struct b {
  static int value;
};

template <class>
using c = disjunction<b>;

auto c_v = c<b>::value;

} // namespace std
```
clang-tidy-17 and current trunk with `--checks=modernize-type-traits --extra-arg=-std=c++17` args produce
```
<source>:13:12: warning: use c++17 style variable templates [modernize-type-traits]
   13 | auto c_v = c<b>::value;
      |            ^   ~~~~~~~
      |             _v
1 warning generated.
```
In other words, it advises to use `c_v` in definition of c_v. Renaming `std` namespace or `disjunction` struct or `value` field make the warning go away.

This is problematic if `--warnings-as-errors` is used, I did not manage to think of a way to bypass this bug other than copying the `std::disjunction` implementation to my namespace.

**Comment by xgupta** [comment]
You can also bypass this warning by adding // NOLINT.
``auto c_v = c<b>::value;  // NOLINT``

**Comment by njames93** [comment]
Seems like the check is incorrectly using the desugared type

**Comment by BredPet** [comment]
https://godbolt.org/z/ooPTWG5fT

```c++
#include <type_traits>

template <template <typename> typename CUSTOM_TYPE, typename... T>
inline constexpr bool are_all_t_v = (... && CUSTOM_TYPE<T>::value);

template <class T>
using int_or_double = std::disjunction<std::is_same<T, int>, std::is_same<T, double>>;
template <class T>
constexpr bool int_or_double_v = int_or_double<T>::value;

int main() {
    [[maybe_unused]] auto check1 = are_all_t_v<std::is_integral, int, short>;
    [[maybe_unused]] auto check2 = int_or_double_v<int>;
}
```
produces
```bash
<source>:4:45: warning: use c++17 style variable templates [modernize-type-traits]
    4 | inline constexpr bool are_all_t_v = (... && CUSTOM_TYPE<T>::value);
      |                                             ^             ~~~~~~~
      |                                                        _v
<source>:9:34: warning: use c++17 style variable templates [modernize-type-traits]
    9 | constexpr bool int_or_double_v = int_or_double<T>::value;
      |                                  ^               ~~~~~~~
      |                                               _v
3 warnings generated.
```

It looks like this is still applicable to all clang branches and trunk. It feels as though the falls are favorably directed at logical constructs, like `&&`, `std::conjunction` and `std::disjunction`
